esp/esp/program/modules/handlers/onsitecheckinmodule.py

```diff
@@ -242,23 +242,31 @@ def ajaxbarcodecheckin(self, request, tl, one, two, module, extra, prog):
     @aux_call
     @needs_onsite
     def checkin(self, request, tl, one, two, module, extra, prog):
-        user, found = search_for_user(request, self.program.students_union())
-        if not found:
-            return user
-
-        self.student = user
-
-        if request.method == 'POST':
-            for key in ['attended','paid','liab','med']:
-                if key in request.POST:
-                    self.create_record(key)
-                else:
-                    self.delete_record(key)
+        if request.method == 'POST' and 'userid' in request.POST:
+            error = False
+            message = None
+            user = ESPUser.objects.filter(id = request.POST['userid']).first()
+            if user:
+                self.student = user
+                for key in ['attended','paid','liab','med']:
+                    if key in request.POST:
+                        self.create_record(key)
+                    else:
+                        self.delete_record(key)
+                message = "Check-in updated for " + user.username
+            else:
+                error = True
```

Comment:
Maybe set message here to user not found or something?

Reply:
That's the message that gets displayed when `error = True`:
https://github.com/learning-unlimited/ESP-Website/blob/9000414a8d60722cd9bf0483c1f9b0da9f675d5f/esp/templates/users/usersearch.html#L25-L27